src/Plugins/RpcServer/RpcServer.SmartContract.cs

```diff
@@ -212,81 +212,122 @@ private static Witness[] WitnessesFromJson(JArray _params)
             }).ToArray();
         }
 
-        [RpcMethod]
-        protected internal virtual JToken InvokeFunction(JArray _params)
+        /// <summary>
+        /// Invokes a smart contract with its scripthash based on the specified operation and parameters and returns the result.
+        /// </summary>
+        /// <remarks>
+        /// This method is used to test your VM script as if they ran on the blockchain at that point in time.
+        /// This RPC call does not affect the blockchain in any way.
+        /// </remarks>
+        /// <param name="scriptHash">Smart contract scripthash. Use big endian for Hash160, little endian for ByteArray.</param>
+        /// <param name="operation">The operation name (string)</param>
+        /// <param name="args">Optional. The parameters to be passed into the smart contract operation</param>
+        /// <param name="signers">Optional. List of contract signature accounts.</param>
+        /// <param name="useDiagnostic">Optional. Flag to enable diagnostic information.</param>
+        /// <returns>A JToken containing the result of the invocation.</returns>
+        [RpcMethodWithParams]
+        protected internal virtual JToken InvokeFunction(string scriptHash, string operation, ContractParameter[] args = null, Signer[] signers = null, bool useDiagnostic = false)
```

Comment:
This behaviour diverges from an old one. This method should accept the list of signers with witnesses as the fourth parameter, not only the list of signers.

Reply:
It never worked. And not bing of any use at anywhere.

Reply:
> And not bing of any use at anywhere.

You don't know that. It's an external API and it's a _very_ widely used one. Your changes can easily break real applications. Refactorings _must not_ break compatibility.